### PR TITLE
Enhacements

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

- [x] Add psalm fixed tests.
- [x] Remove closure config.
- [x] Remove composer-plugin tests.
- [x] Update workflows.
- [x] Update REAME.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ It helps resolving dependencies and getting lists of files ready for generating
 [![Code Coverage](https://scrutinizer-ci.com/g/yiisoft/assets/badges/coverage.png?b=master)](https://scrutinizer-ci.com/g/yiisoft/assets/?branch=master)
 [![Mutation testing badge](https://img.shields.io/endpoint?style=flat&url=https%3A%2F%2Fbadge-api.stryker-mutator.io%2Fgithub.com%2Fyiisoft%2Fassets%2Fmaster)](https://dashboard.stryker-mutator.io/reports/github.com/yiisoft/assets/master)
 [![static analysis](https://github.com/yiisoft/assets/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/assets/actions?query=workflow%3A%22static+analysis%22)
+[![type-coverage](https://shepherd.dev/github/yiisoft/assets/coverage.svg)](https://shepherd.dev/github/yiisoft/assets)
```
Comment (samdark) [inline README.md:19]: Type coverage?

Comment (terabytesoftw) [reply README.md:19]: If that's the psalm coverage of static analysis, I think that's a good metric to take into account.

File: config/web.php [hunk 2]
```diff
@@ -2,81 +2,49 @@
...
-
-        if ($params['yiisoft/asset']['assetManager']['register'] !== []) {
-            $assetManager->register($params['yiisoft/asset']['assetManager']['register']);
-        }
-
-        return $assetManager;
-    },
+    AssetConverterInterface::class => [
+        '__class' => AssetConverter::class,
+        '__construct()' => [
```
Comment (samdark) [inline config/web.php:18]: Won't these be resolved automatically?

File: config/web.php [hunk 3]
```diff
@@ -2,81 +2,49 @@
...
+            $params['yiisoft/asset']['assetConverter']['command']['from'],
+            $params['yiisoft/asset']['assetConverter']['command']['to'],
+            $params['yiisoft/asset']['assetConverter']['command']['command']
+        ],
+        'setForceConvert()' => [$params['yiisoft/asset']['assetConverter']['forceConvert']]
+    ],
+
+    AssetPublisherInterface::class => [
+        '__class' => AssetPublisher::class,
+        '__construct()' => [Reference::to(Aliases::class)],
```
Comment (samdark) [inline config/web.php:32]: Won't these be resolved automatically?

File: config/web.php [hunk 4]
```diff
@@ -2,81 +2,49 @@
...
+        'setBasePath()' => $params['yiisoft/asset']['assetPublisher']['basePath'],
+        'setBaseUrl()' => $params['yiisoft/asset']['assetPublisher']['baseUrl'],
+        'setForceCopy()' => $params['yiisoft/asset']['assetPublisher']['forceCopy'],
+        'setLinkAssets()' => $params['yiisoft/asset']['assetPublisher']['linkAssets']
+
+    ],
+
+    AssetManager::class => [
+        '__class' => AssetManager::class,
+        '__construct()' => [Reference::to(LoggerInterface::class)],
```
Comment (samdark) [inline config/web.php:44]: Won't these be resolved automatically?

File: tests/TestCase.php [hunk 5]
```diff
@@ -145,4 +147,57 @@ protected function unlink(string $file): bool
...
+                'setBasePath()' => [$params['yiisoft/asset']['assetPublisher']['basePath']],
+                'setBaseUrl()' => [$params['yiisoft/asset']['assetPublisher']['baseUrl']],
+                'setForceCopy()' => [$params['yiisoft/asset']['assetPublisher']['forceCopy']],
+                'setLinkAssets()' => [$params['yiisoft/asset']['assetPublisher']['linkAssets']]
+
+            ],
+
+            AssetManager::class => [
+                '__class' => AssetManager::class,
+                '__construct()' => [Reference::to(LoggerInterface::class)],
```
Comment (samdark) [inline tests/TestCase.php:190]: Won't these be resolved automatically?

File: tests/TestCase.php [hunk 6]
```diff
@@ -145,4 +147,57 @@ protected function unlink(string $file): bool
...
+                    $params['yiisoft/asset']['assetConverter']['command']['from'],
+                    $params['yiisoft/asset']['assetConverter']['command']['to'],
+                    $params['yiisoft/asset']['assetConverter']['command']['command']
+                ],
+                'setForceConvert()' => [$params['yiisoft/asset']['assetConverter']['forceConvert']]
+            ],
+
+            AssetPublisherInterface::class => [
+                '__class' => AssetPublisher::class,
+                '__construct()' => [Reference::to(Aliases::class)],
```
Comment (samdark) [inline tests/TestCase.php:178]: Won't these be resolved automatically?

File: tests/TestCase.php [hunk 7]
```diff
@@ -145,4 +147,57 @@ protected function unlink(string $file): bool
...
+        return [
+            Aliases::class => [
+                '__construct()' => [$params['aliases']]
+            ],
+
+            LoggerInterface::class => NullLogger::class,
+
+            AssetConverterInterface::class => [
+                '__class' => AssetConverter::class,
+                '__construct()' => [
```
Comment (samdark) [inline tests/TestCase.php:164]: Won't these be resolved automatically?

**Review by samdark — CHANGES_REQUESTED** [review]

File: config/web.php (hunk 2 above)
Comment (terabytesoftw) [reply config/web.php]: done

File: config/web.php (hunk 3 above)
Comment (terabytesoftw) [reply config/web.php]: done.

File: config/web.php (hunk 4 above)
Comment (terabytesoftw) [reply config/web.php]: done

File: tests/TestCase.php (hunk 5 above)
Comment (terabytesoftw) [reply tests/TestCase.php:190]: done.

File: tests/TestCase.php (hunk 6 above)
Comment (terabytesoftw) [reply tests/TestCase.php:178]: done.

File: tests/TestCase.php (hunk 7 above)
Comment (terabytesoftw) [reply tests/TestCase.php:164]: done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**